src/compile.ts

```diff
@@ -58,21 +58,28 @@ export function compile(
         if (isAppsyncResolver(node)) {
           return visitAppsyncResolver(node);
         } else if (isReflectFunction(node)) {
-          return toFunction("FunctionDecl", node.arguments[0]);
+          const func = node.arguments[0];
+          if (ts.isIdentifier(func)) {
+            throw new Error(
+              `Reflect does not support function references: ${func.text}`
```

Comment:
Maybe also add, "please pass an inline function to reflect".

Reply:
Also, this requirement isn't only for reflect. It should be for anywhere we convert a function into its AST form.

Reply:
Will update the other places to reflect this, also I think a few other statements could act as a reference like Call or property access as you mention in the document. 

How much did you looking into resolving the references as opposed to needing to be inline? (I really want to pass the arrow functions to a test method, but ran into this).

Reply:
I hope we could solve it also - I think this is part of a larger problem of "closure serialization". For example, we would also want to be able to serialize a Lambda Function implementation, including all of the data it captures in the closure and any dependent functions.

```ts
import dep from "some-module";

class Service extends Construct {
  constructor(scope: Construct, id: string, props: Props) {
    new LambdaFunction(this, "Func", (name: string) => {
      return dep.someFunction(name, props.someCapturedClosureData);      
    });
  }
}
```

The ideal state is that all valid NodeJS applications are valid Functionless applications.